File: crud-item-list/src/main/java/com/gurunars/crud_item_list/ListChangeListener.java
package com.gurunars.crud_item_list;

import com.gurunars.item_list.Item;

import java.util.List;

/**
 * Listener to be called whenever the collection manipulated by the widget is modified.
 *
 * @param <ItemType> class of the items managed by the widget.
 */
public interface ListChangeListener<ItemType extends Item> {

    /**
     * @param items a new version ot the list to be saved
     */
    void onChange(List<ItemType> items);

    class DefaultListChangeListener<ItemType extends Item> implements ListChangeListener<ItemType> {

        @Override
        public void onChange(List<ItemType> items) {

        }
    }

}

File: crud-item-list/src/main/kotlin/com/gurunars/crud_item_list/ListChangeListener.kt
package com.gurunars.crud_item_list

import com.gurunars.item_list.Item

/**
 * Listener to be called whenever the collection manipulated by the widget is modified.

 * @param <ItemType> class of the items managed by the widget.
</ItemType> */
interface ListChangeListener<ItemType : Item> {

    /**
     * @param items a new version ot the list to be saved
     */
    fun onChange(items: List<ItemType>)

    class DefaultListChangeListener<ItemType : Item> : ListChangeListener<ItemType> {

        override fun onChange(items: List<ItemType>) {

        }
    }

}
